src/services/pasteEdits.ts

```diff
@@ -77,7 +76,19 @@ function pasteEdits(
         if (copiedFrom?.range) {
             Debug.assert(copiedFrom.range.length === pastedText.length);
             copiedFrom.range.forEach(copy => {
-                addRange(statements, copiedFrom.file.statements, getLineOfLocalPosition(copiedFrom.file, copy.pos), getLineOfLocalPosition(copiedFrom.file, copy.end) + 1);
+                const statementsInSourceFile = copiedFrom.file.statements;
+                const startNodeIndex = findIndex(statementsInSourceFile, s => s.end > copy.pos);
+                if (startNodeIndex === -1) return undefined;
+                let endNodeIndex = findIndex(statementsInSourceFile, s => s.end >= copy.end, startNodeIndex);
+                /**
+                 * [|console.log(a);
+                 * |]
+                 * console.log(b);
+                 */
+                if (endNodeIndex !== -1 && copy.end <= statementsInSourceFile[endNodeIndex].getStart()) {
+                    endNodeIndex--;
+                }

```

Comment:
I think this could be baked into the `findIndex` predicate, but this is alright.